### PR TITLE
7 post comments

### PR DESCRIPTION
**Pull request by vh232**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -27,174 +27,280 @@ describe("GET /api/topics", () => {
...
+      body: "best article ever",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("user does not exist");
+      });
+  });
```
Comment (MKD01) [inline __tests__/app.test.js]: Should this be a 400 status code? if the user vh232 does not exist does that make them an invalid user or a user that does not exist?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -27,174 +27,280 @@ describe("GET /api/topics", () => {
...
+      body: "best article ever",
+      extraKey: "whoops"
+    }
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("incorrect format");
       })
```
Comment (MKD01) [inline __tests__/app.test.js]: It doesnt seem like the newComment2 is being used here, and there is a test below which does test for this case so I dont think this object needs to be here

File: __tests__/app.test.js [hunk 3]
```diff
@@ -27,174 +27,280 @@ describe("GET /api/topics", () => {
...
-      const { articles } = body
-      expect(articles.length).toBe(13)
-      articles.forEach((article) => {
-        expect(article).not.toHaveProperty('body')
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("incorrect format");
       })
```
Comment (MKD01) [inline __tests__/app.test.js]: This is a very important test to have, but should this return an error if given extra unnecessary keys? Is it incorrect to give extra keys in this case, as long as the necessary keys are given and they are valid?

File: endpoints.json [hunk 4]
```diff
@@ -34,5 +34,40 @@
...
+      "created_at": 1604113380000
+    },
+    {
+      "body": "Replacing the quiet elegance of the dark suit and tie with the casual indifference of these muted earth tones is a form of fashion suicide, but, uh, call me crazy — onyou it works.",
+      "votes": 100,
+      "author": "icellusedkars",
+      "article_id": 1,
+      "created_at": 1583025180000
+    }]
   }
```
Comment (MKD01) [inline endpoints.json:88]: Dont forget to add the post endpoint into this endpoints.json file as well

**Review by MKD01 — CHANGES_REQUESTED** [review]
Really good work so far, but I would like to see the changes made to the tests before you merge this in, so please re-submit this pr once you have made those changes.

**Review by aloush-dev — APPROVED** [review]
Awesome job implementing feedback!

Mergeeeee.

**Pull request merged**